File: lib/src/test/java/io/xj/lib/music/OctaveTest.java
// Copyright (c) XJ Music Inc. (https://xjmusic.com) All Rights Reserved.
package io.xj.lib.music;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class OctaveTest {

  @Test
  public void of() throws Exception {
    assertEquals(Integer.valueOf(0), Octave.of("F"));
    assertEquals(Integer.valueOf(2), Octave.of("Bb2"));
    assertEquals(Integer.valueOf(3), Octave.of("D#3"));
    assertEquals(Integer.valueOf(5), Octave.of("D5"));
    assertEquals(Integer.valueOf(-2), Octave.of("D-2"));
    assertEquals(Integer.valueOf(-2), Octave.of("D--2"));
  }

}

File: engine/test/music/OctaveTest.cpp
// Copyright (c) XJ Music Inc. (https://xjmusic.com) All Rights Reserved.

#include <gtest/gtest.h>

#include "xjmusic/music/Octave.h"

using namespace Music;

TEST(Music_Octave, of) {
  ASSERT_EQ(0, octaveOf("F"));
  ASSERT_EQ(2, octaveOf("Bb2"));
  ASSERT_EQ(3, octaveOf("D#3"));
  ASSERT_EQ(5, octaveOf("D5"));
  ASSERT_EQ(-2, octaveOf("D-2"));
  ASSERT_EQ(-2, octaveOf("D--2"));
}
